lib/editor-linter.coffee

```diff
@@ -53,30 +53,34 @@ class EditorLinter
 
     scopes = @editor.scopeDescriptorForBufferPosition(@editor.getCursorBufferPosition()).scopes
     scopes.push '*' # To allow global linters
-
-    Promise.all(@triggerLinters(wasTriggeredOnChange, scopes)).then =>
+    @triggerLinters(true, wasTriggeredOnChange, scopes).then( =>
+      return Promise.all(@triggerLinters(false, wasTriggeredOnChange, scopes))
+    ).then =>
       @lock(wasTriggeredOnChange, false)
 
   # This method returns an array of promises to be used in lint
-  triggerLinters: (wasTriggeredOnChange, scopes) ->
-    Promises = []
+  triggerLinters: (bufferModifying, wasTriggeredOnChange, scopes) ->
+    ToReturn = if bufferModifying then Promise.resolve() else []
     @linter.getLinters().forEach (linter) =>
```

Comment:
does `getLinters()` return an Array? Right now this whole `ToReturn` thing is really just simulating a `.map()` or a `.reduce()` depending on if the linters modify the buffer. If you can I recommend `return @linter.getLinters().reduce(...)` instead